src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -940,7 +941,8 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
         // We don't support implementation_buffer implementations that use nextimpl_working_buffer_offset.
         chassert(implementation_buffer->position() == implementation_buffer->buffer().begin());
 
-        size = implementation_buffer->buffer().size();
+        if (result)
+            size = implementation_buffer->buffer().size();
```

Comment:
If `!result` , then previous value will be used in the log & incrementing ProfileEvents too. Is this fine ?
https://github.com/ClickHouse/ClickHouse/blob/a1eb27ce3038df0921d6667e65074168776988dc/src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp#L900